docgen/src/guide/Server-side_rendering.md

```diff
@@ -6,8 +6,24 @@ category: guide
 navWeight: 30
 ---
 
-React InstantSearch is not yet compatible with server-side rendering. This is one of our priority topics
-right now. We will soon provide API entries to handle this usecase.
+React InstantSearch builds the Algolia query based on every widget declared. That's why at the first rendering there are no results yet. This is inconvenient when doing server side rendering as the html generated by the server will not contain any hits. 
```

Comment:
I feel like we should start by saying "React InstantSearch is compatible with server-side rendering, here are examples on how to use it:".

Then at the end of the page add a paragraph explaining a bit the inner details like this first sentence.

Otherwise I open this page, I see a sentence that seem negative, I think "well maybe doesn't work, let's move on".